crates/sui-indexer/src/store/pg_indexer_store.rs

```diff
@@ -2187,38 +2145,49 @@ fn make_objects_history_to_commit(
     deleted_objects.into_iter().chain(mutated_objects).collect()
 }
 
-fn make_objects_snapshot_to_commit(
+// Partition object changes into deletions and mutations,
+// within partition of mutations or deletions, retain the latest with highest version;
+// For overlappings of mutations and deletions, only keep one with higher version.
+// This is necessary b/c after this step, DB commit will be done in parallel and not in order.
+fn retain_latest_indexed_objects(
     tx_object_changes: Vec<TransactionObjectChangesToCommit>,
-) -> Vec<StoredObjectSnapshot> {
-    let deleted_objects = tx_object_changes
-        .clone()
+) -> (Vec<IndexedObject>, Vec<IndexedDeletedObject>) {
+    // Only the last deleted / mutated object will be in the map,
+    // b/c tx_object_changes are in order and versions always increment,
```

Comment:
This function talks about retaining the one with the highest version but it doesn't seem like there are any comparisons against the version of each object. I assume this is the reason why? Do we want to have some logic that does a check?

Reply:
yes exactly, I started with the comparison approach on the other similar fix and @lxfind suggested this which seems a bit cleaner https://github.com/MystenLabs/sui/pull/18991/files#r1723999314 

and sure, I can add some asserting logic here to make sure.